shared/aux.go

```diff
@@ -325,3 +326,59 @@ func fileExists(files []os.DirEntry, workload string) bool {
 
 	return false
 }
+
+func EncloseInBrackets(ip string)string {
+	return "["+ip+"]"
+}
+
+func GetNodeArgs(nodeType string) (map[string]string, error){
+	product,err := GetProduct()
+	if err != nil{
+		return nil, err
+	}
+	res,err := KubectlCommand(
+		"host",
+		"get",
+		"nodes " + 
+		fmt.Sprintf(`-o jsonpath='{range .items[*]}{.metadata.annotations.%s\.io/node-args}{end}'`, product),
+	)
+	if err != nil{
+		return nil, err
+	}
+
+	argsMapSlice := processNodeArgsResponse(res)
+
+	for _, argsMap := range argsMapSlice {
+		if argsMap["node-type"] == nodeType {
+			return argsMap, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func processNodeArgsResponse(res string) ([]map[string]string){
+	nodeArgsMapSlice := []map[string]string{}
+	resSlice := strings.Split(res,"]")
+	
+	for _,resItem := range(resSlice[:(len(resSlice)-1)]){
+		resItemSlice := strings.Split(resItem,`","`)
+		nodeArgsMap := map[string]string{}
+
+		for range(resItemSlice[1:]){
```

Comment:
you probably dont need this for resitemslice

Reply:
I need to iterate starting from second item and skip the first item in the slice, is it not how it should be done?

Reply:
those assign 
nodeArgsMap["node-type"] = strings.Trim(resItemSlice[0],`["`)
			regxCompile := regexp.MustCompile(`--|"`)

u probably can done only once since in all iterations of 2 second loop the resItemSlice[0]  , does not change.

Reply:
try remove 

for range(resItemSlice[1:]){.  and the close }